### DISCUSSION TITLE
Questions about database Seeds and Migrations

### DISCUSSION THREAD

**Discussion by lirantal** [prisma/prisma#16247] [Q&A] [answered]
Hi folks,

I'm looking into Prisma and the docs have fallen short to help me figure out a few key questions I have as I evaluate using Prisma:
1. It seems that migrations are only scoped to the actual schema structure and cannot hold any record changes, such as adding, updating or deleting data?
2. It seems that migrations are strictly made to one file (`schema.js`) and when migrations are to be executed it creates specified timestamped migration files. So, all and any changes to the database structure always remains in schema.js?
3. Similarly to the migration file above, seeding seems to be only done in the `prisma/seed.js` file but in that case is it a one-off single file run? how do I make more seeded changes (changing data in the database) than just once?

**Comment by pantharshit00** [comment, marked as answer]
Hello @lirantal,

> 1. It seems that migrations are only scoped to the actual schema structure and cannot hold any record changes, such as adding, updating or deleting data?

You can edit the migrations to store any SQL statement you want. You will use the `--create-only` flag and then add the changes you want. Read more: https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations


> 2. It seems that migrations are strictly made to one file (schema.js) and when migrations are to be executed it creates specified timestamped migration files. So, all and any changes to the database structure always remains in schema.js?

I think you want to refer the `schema.prisma` file here. There is no specific `schema.js` used by Prisma. Yes, `schema.prisma` file reflects the current database structure and migrations are created by diffing the database with the schema file. Then those migrations are stored in `prisma/migrations` folder. You can also pull the current database structure in the Prisma Schema Format using the `db pull` command: https://www.prisma.io/docs/reference/api-reference/command-reference#db-pull

> Similarly to the migration file above, seeding seems to be only done in the prisma/seed.js file but in that case is it a one-off single file run? how do I make more seeded changes (changing data in the database) than just once?

The seed script workflow that we provide is meant for development where you can incur dataloss and reseed the data. If you want to edit production data, either you can store the SQL statements in the migrations file by editing them as I shared in the first point or you will need to store a script yourself which you will need to run manually.

**Reply by lirantal** [reply]
About (1), the first step in customizing the migrations state that some sort of schema change is still required:
> Make a schema change that requires custom SQL (for example, to preserve existing data)

What if I don't have schema changes to make? just want to seed a new set of data for config, dropdown, etc? I guess I can make some one-off change and then revert it with adding the rest of the SQL I want but it feels somewhat odd.

Anyway I think that answers all of my questions. Thanks.

**Reply by pantharshit00** [reply]
> What if I don't have schema changes to make? just want to seed a new set of data for config, dropdown, etc? I guess I can make some one-off change and then revert it with adding the rest of the SQL I want but it feels somewhat odd.

`--create-only` flag also allow you to create empty migrations where you can add SQL statements which you want.

**Reply by lirantal** [reply]
> You can edit the migrations to store any SQL statement you want. You will use the --create-only flag and then add the changes you want. Read more: https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/customizing-migrations

@pantharshit00 aaaaah that totally makes sense now. I missed this part of the docs.

Thank you!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
